### ISSUE TITLE
Sourcemap path in current bundle isn't being formed properly

### ISSUE THREAD

**Issue by JSMike** [angular/angular#14021]
**I'm submitting a ...**  (check one with "x")
```
[ x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
Sourcemaps are being added to each angular bundle from the perspective of the wrong directory. These sourcemaps are currently included in the latest (2.4.4) release when doing a fresh npm install.

Example from `node_modules/@angular/core/src/linker/compiler.js.map`

```
{"version":3,"file":"compiler.js","sourceRoot":"","sources":["../../../../../modules/@angular/core/src/linker/compiler.ts"],"names":[],
```

This is causing an unexpected/incorrect path to be displayed in debugging tools.

**Expected behavior**
Sourcemap paths should start from the npm repository's base directory. In this case the path should start from `modules/@angular/core`

Resulting in a path similar to this:

```
{"version":3,"file":"compiler.js","sourceRoot":"","sources":["src/linker/compiler.ts"],"names":[],

```

**Minimal reproduction of the problem with instructions**

Create a new project with Angular CLI beta.25-5

modify the file: node_modules/angular-cli/models/webpack-build-common.js:132

replace `exclude: [nodeModules]` with `exclude: /node_modules\/(?!(\@angular\/core)).*/`

*note: this is just to quickly include sourcemaps for @angular/core

run `ng serve`

Open the served page in Chrome and use developer tools > Sources tab
As a result (when using webpack) files appear to be in:

```
webpack://../../../../../modules/@angular/core/src/linker/compiler.ts
```

instead of 
```
webpack://./~/@angular/core/src/linker/compiler.ts
```

**What is the motivation / use case for changing the behavior?**
Debugging Angular codebase from the context of a CLI project (I'm assuming whenever using webpack).

**Please tell us about your environment:**

OSX, VSCode, NPM, Webpack Dev Sever

* **Angular version:** 2.0.X

2.4.4

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

 Chrome, I haven't tried others

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
TypeScript

* **Node (for AoT issues):** `node --version` =   
angular-cli: 1.0.0-beta.25.5
node: 6.5.0
os: linux x64
@angular/common: 2.4.4
@angular/compiler: 2.4.4
@angular/core: 2.4.4
@angular/forms: 2.4.4
@angular/http: 2.4.4
@angular/material: 2.0.0-beta.1
@angular/platform-browser: 2.4.4
@angular/platform-browser-dynamic: 2.4.4
@angular/router: 3.4.4
@angular/compiler-cli: 2.4.4

Also note, I believe that webpack may be having issues with the sourcemaps plugin for scoped npm packages (anything @username/project) it's not placing the sources in the correct location even when the sourcemaps are corrected.

**Comment by JSMike** [comment]
it looks like the tsconfig-build.json file is missing the compilerOption `"mapRoot": "./"`

**Comment by JSMike** [comment]
I did some digging, it has to do with `tools/@angular/tsc-wrapped/src/main.ts`.
specifically:
 * the cwd() / projectDir / basePath when main.ts is called by build.sh ( ${TSC} ), 
 * the destination of the build files.
 * the values of the rootDir, mapRoot, (and maybe baseUrl?) in the tsconfig-build.json for each module

I tried a few combinations but I didn't have much luck. If I can find more time to work on this I will, but I'm very busy the next couple of weeks. If someone else wants to give it a shot, you can use the following command from the package root and attach a listener in vscode to step through the process.

```
node --max-old-space-size=3000 --debug-brk dist/tools/@angular/tsc-wrapped/src/main -p `pwd`/modules/@angular/core/tsconfig-build.json
```
the output files can then be found in `dist/packages-dist/core`

**Comment by DzmitryShylovich** [comment]
@JSMike is this still valid in ng4?

**Comment by JSMike** [comment]
@DzmitryShylovich yes, this is still an issue. It actually looks significantly worse in ng4 since the files being used are the bundled .es5.js ones.

I've been digging into this a bit when I have free time, there are ways to get it to work with some tsconfig and .map file updates, but so far I've only gotten it to work by including the .ts files with the bundle as well as making changes to tsconfig and the map files. Chrome isn't loading the sourcesContent that's included in the .es5.map.js file as expected.

**Comment by JSMike** [comment]
Having issues looking for a solution right now because of this: https://github.com/angular/angular/issues/15527 when trying to use `ngc`

**Comment by IgorMinar** [comment]
somewhat related issue: #19841

**Comment by JoostK** [comment]
This is definitely still an issue, although the source path is slightly different:

- [@angular/core@6.0.0](https://unpkg.com/browse/@angular/core@6.0.0/fesm2015/core.js.map)
  - Very much incorrect, including Bazel output paths.
- [@angular/core@7.0.0 & 8.0.0](https://unpkg.com/browse/@angular/core@8.0.0/fesm2015/core.js.map)
  - These look correct to me, although they are relative to the map file and not the NPM root as this issue suggest they should be. I don't know whether this is an issue.
- [@angular/core@9.0.0](https://unpkg.com/browse/@angular/core@9.0.0/fesm2015/core.js.map)
  - We're back at an incorrect relative path, which seems worse compared to 8.0

@devversion are you aware of any change from 8.0 to 9.0 in the Bazel stuff that could have caused the change?

**Comment by devversion** [comment]
@JoostK I'm not aware of any change from v8 to v9 that could cause these invalid source map paths. 

I checked whether 958165888c caused any regression (sine it's path generation related), but looks like it didn't change anything unexpectedly. https://github.com/angular/core-builds/commit/f4cf44d57c5c16308040780d53aee8e67cf216c9. I can double-check if I find something, but I remember some source map paths being broken for quite some while (even in v8 IIRC) when I created the size-tracking tool.

**Comment by IgorMinar** [comment]
This is still broken in [@angular/core@10.0.0-rc.0](https://unpkg.com/browse/@angular/core@10.0.0-rc.0/fesm2015/core.js.map)

**Comment by petebacondarwin** [comment]
Keeping this as the primary tracking issue for problems related to the `sources` paths in the source-maps generated by the Angular build. Related issues that are tracked here:

* Sourcemaps for .d.ts files break go to definition in vscode #34446
* Bazel - Relocate sourcemaps #23520
* NGCC is non-hermetic and produces different output depending on source location in file system #39293

**Comment by JeanMeche** [comment]
Without further reports in the last 5 years, I'll consider this issue closed.
